Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -141,46 +133,66 @@ class UIRedactBuilder {
+        
+        let layer = view.layer.presentation() ?? view.layer
+        let size = layer.bounds.size
+        let layerMiddle = CGPoint(x: size.width * layer.anchorPoint.x, y: size.height * layer.anchorPoint.y)
+        
+        var newTransform = transform.translatedBy(x: layer.position.x, y: layer.position.y)
+        newTransform = view.transform.concatenating(newTransform)
+        newTransform = newTransform.translatedBy(x: -layerMiddle.x, y: -layerMiddle.y)
         
         let ignore = shouldIgnore(view: view)
-        let redact = shouldRedact(view: view, redactText: redactText, redactImage: redactImage)
+        let redact = shouldRedact(view: view, redactOptions: redactOptions)
         
         if !ignore && redact {
-            redacting.append(RedactRegion(rect: rectInWindow, color: self.color(for: view)))
+            redacting.append(RedactRegion(size: size, transform: newTransform, type: .redact, color: self.color(for: view)))
             return
-        } else if hasBackground(view) {
-            if rectInWindow == area {
+        } else if isOpaque(view) {
+            let finalViewFrame = CGRect(origin: .zero, size: size).applying(newTransform)
+            if isAxisAligned(newTransform) && finalViewFrame == rootFrame {
+                //Because the current view is covering everything we found so far we can clear `redacting` list
                 redacting.removeAll()
             } else {
-                redacting = redacting.flatMap { $0.splitBySubtracting(region: rectInWindow) }
+                redacting.append(RedactRegion(size: size, transform: newTransform, type: .clip))
             }
         }
         
         if !ignore {
             for subview in view.subviews {
-                mapRedactRegion(fromView: subview, to: to, redacting: &redacting, area: area, redactText: redactText, redactImage: redactImage)
+                mapRedactRegion(fromView: subview, redacting: &redacting, rootFrame: rootFrame, redactOptions: redactOptions, transform: newTransform)
             }
         }
     }
     
+    /**
+     Whether the transform does not contains rotation or skew
+     */
+    func isAxisAligned(_ transform: CGAffineTransform) -> Bool {
+        // Rotation exists if b or c are not zero
+        return transform.b == 0 && transform.c == 0
+    }
+
     private func color(for view: UIView) -> UIColor? {
         return (view as? UILabel)?.textColor
     }
     
-    private func hasBackground(_ view: UIView) -> Bool {
+    /**
+     Indicates whether the view is opaque and will block other view behind it
+     */
+    private func isOpaque(_ view: UIView) -> Bool {
         //Anything with an alpha greater than 0.9 is opaque enough that it's impossible to see anything behind it.
-        return view.backgroundColor != nil && (view.backgroundColor?.cgColor.alpha ?? 0) > 0.9
+        return  view.alpha > 0.9 && view.backgroundColor != nil && (view.backgroundColor?.cgColor.alpha ?? 0) > 0.9
```

Comment:
How did we arrive at this conclusion? I tested with a view with background color white alpha 0.9 and it reports it as opaque and doesn't redact the text beneath it: 


I added a log just to be sure what the code was doing:
```
View with color alpha Optional(0.9111128827) considered opaque
```

![image](https://github.com/user-attachments/assets/b73eccb5-5a4b-4044-8b88-8a5160be8618)

Reply:
This is an arbitrary number. If we believe this is not good enough we can raise the number.

Reply:
Changing it to 1. Although 0.99 is non human-readable, it’s possible to use image processing to figure out the information behind it.

Reply:
My thought exactly. Thanks for updating.